File: problems-extend-adapt/0801-0900/0810_xor-erasure-duel/solution.py
from typing import List


class Solution:
    def firstPlayerWins(self, nums: List[int]) -> bool:
        # Alice wins exactly when the board already folds to 0 (she wins
        # on the spot) or the count is even, letting her always hand Bob
        # a nonzero odd board he cannot escape.
        xor = 0
        for value in nums:
            xor ^= value
        return xor == 0 or len(nums) % 2 == 0

File: problems-extend-adapt/0801-0900/0810_xor-erasure-duel/solution.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var firstPlayerWins = function (nums) {
    // Alice wins exactly when the board already folds to 0 (she wins
    // on the spot) or the count is even, letting her always hand Bob
    // a nonzero odd board he cannot escape.
    let total = 0;
    for (const value of nums) {
        total ^= value;
    }
    return total === 0 || nums.length % 2 === 0;
};
